src/pages/Timeline/CreatePost.jsx

```diff
@@ -0,0 +1,184 @@
+import styled from 'styled-components';
+import React from 'react';
+import { useState, useContext } from 'react';
+import Swal from 'sweetalert2';
+import { createPostAPI } from '../../service/service.posts';
+import UserContext from '../../contexts/UserContext';
+
+export default function CreatePost() {
+	const {userInfo} = useContext(UserContext);
+	const [link, setLink] = useState('');
+	const [linkDescription, setLinkDescription] = useState('');
+	const [loading, setLoading] = useState(0);
+
```

Comment:
Aqui faria mais sentido termos um booleano no estado. Colocar 0 e 1 pode dar a entender que podemos colocar qualquer outro número nesse estado.

Reply:
então, eu tinha feito como booleano, mas por algum motivo eu recebia warning quando passava um booleano como parametro pros styled components. Só para retirar essa warning eu mudei para 0/1

Reply:
Colocar booleano também implica em um cuidado a mais se você for passar isso como props pra algum componente. Ao invés de usar loading={loading}, você tem que usar loading={loading ? 1 : 0}